src/app/admin/packingSlotsTable/PackingSlotsTable.tsx

```diff
@@ -0,0 +1,332 @@
+
+    const handleUpClick = (id: GridRowId, row: PackingSlotRow) => () => {
+        const rowIndex = row.order - 1;
+        if (rowIndex > 0) {
+            setIsLoading(true);
+            const rowOne = rows[rowIndex];
+            const rowTwo = rows[rowIndex - 1];
+            swapRows(rowOne, rowTwo)
+                .catch((error) => void logError("Upsert error with packing slot row order", error))
+                .finally(() => setIsLoading(false));
+        }
+        setIsLoading(false);
+    };
+
+    const handleDownClick = (id: GridRowId, row: PackingSlotRow) => () => {
+        const rowIndex = row.order - 1;
+        if (rowIndex < rows.length - 1) {
+            setIsLoading(true);
+            const clickedRow = rows[rowIndex];
+            const rowBelow = rows[rowIndex + 1];
+            swapRows(clickedRow, rowBelow)
+                .catch((error) => void logError("Upsert error with packing slot row order", error))
+                .finally(() => setIsLoading(false));
+        }
+    };
+
+    const packingSlotsColumns: GridColDef[] = [
+        {
+            field: "order",
+            type: "actions",
+            headerName: "Order",
+            width: 100,
+            cellClassName: "actions",
+            getActions: ({ id, row }) => {
+                if (row.order === 1) {
+                    return [
+                        <GridActionsCellItem
+                            icon={<ArrowCircleDownIcon />}
+                            label="Down"
+                            onClick={handleDownClick(id, row)}
+                            color="inherit"
+                            key="Down"
+                        />,
+                    ];
+                } else if (rows && row.order === rows.length) {
+                    return [
+                        <GridActionsCellItem
+                            icon={<ArrowCircleUpIcon />}
+                            label="Up"
+                            className="textPrimary"
+                            onClick={handleUpClick(id, row)}
+                            color="inherit"
+                            key="Up"
+                        />,
+                    ];
+                } else {
+                    return [
+                        <GridActionsCellItem
+                            icon={<ArrowCircleUpIcon />}
+                            label="Up"
+                            className="textPrimary"
+                            onClick={handleUpClick(id, row)}
+                            color="inherit"
+                            key="Up"
+                        />,
+                        <GridActionsCellItem
+                            icon={<ArrowCircleDownIcon />}
+                            label="Down"
+                            onClick={handleDownClick(id, row)}
+                            color="inherit"
+                            key="Down"
+                        />,
+                    ];
+                }
```

Comment:
If we're returning in the if statement, we don't need an else so we can simply the expression a little bit

```suggestion
                if (row.order === 1) {
                    return [
                        <GridActionsCellItem
                            icon={<ArrowCircleDownIcon />}
                            label="Down"
                            onClick={handleDownClick(id, row)}
                            color="inherit"
                            key="Down"
                        />,
                    ];
                }
                
                if (rows && row.order === rows.length) {
                    return [
                        <GridActionsCellItem
                            icon={<ArrowCircleUpIcon />}
                            label="Up"
                            className="textPrimary"
                            onClick={handleUpClick(id, row)}
                            color="inherit"
                            key="Up"
                        />,
                    ];
                }
                
                return [
                    <GridActionsCellItem
                        icon={<ArrowCircleUpIcon />}
                        label="Up"
                        className="textPrimary"
                        onClick={handleUpClick(id, row)}
                        color="inherit"
                        key="Up"
                    />,
                    <GridActionsCellItem
                        icon={<ArrowCircleDownIcon />}
                        label="Down"
                        onClick={handleDownClick(id, row)}
                        color="inherit"
                        key="Down"
                    />,
                ];
```